collector/processor/concurrentbatchprocessor/batch_processor_test.go

```diff
@@ -323,11 +336,167 @@ func TestBatchProcessorCancelContext(t *testing.T) {
+	require.NoError(t, err)
+
+	bp, err := newBatchTracesProcessor(creationSet, next, cfg, true)
+	require.NoError(t, err)
+	require.NoError(t, bp.Start(context.Background(), componenttest.NewNopHost()))
+
+	sentResourceSpans := ptrace.NewTraces().ResourceSpans()
+	var wg sync.WaitGroup
+	for requestNum := 0; requestNum < requestCount; requestNum++ {
+		td := testdata.GenerateTraces(spansPerRequest)
+		spans := td.ResourceSpans().At(0).ScopeSpans().At(0).Spans()
+		for spanIndex := 0; spanIndex < spansPerRequest; spanIndex++ {
+			spans.At(spanIndex).SetName(getTestSpanName(requestNum, spanIndex))
+		}
+		td.ResourceSpans().At(0).CopyTo(sentResourceSpans.AppendEmpty())
+		// ConsumeTraces is a blocking function and should be run in a go routine
+		// until batch size reached to unblock.
+		wg.Add(1)
+		go func() {
+			assert.NoError(t, bp.ConsumeTraces(bg, td))
+			wg.Done()
+		}()
+	}
+	wg.Wait()
+
+	// need to flush tracerprovider
+	tp.ForceFlush(bg)
+	td := exp.GetSpans()
+
+	numBatches := float64(spansPerRequest*requestCount) / float64(cfg.SendBatchMaxSize)
+	assert.Equal(t, int(math.Ceil(numBatches))+1, len(td))
+	for i := range td {
+		if !td[i].Parent.HasTraceID() {
+			assert.Equal(t, td[i].SpanContext, rootSp.SpanContext())
+			continue
+		}
+		// confirm parent is rootSp
+		assert.Equal(t, td[i].Parent, rootSp.SpanContext())
+	}
+
+	require.NoError(t, bp.Shutdown(context.Background()))
+}
+
+func TestBatchProcessorUnbrokenParentContextMultiple(t *testing.T) {
+	cfg := createDefaultConfig().(*Config)
+	cfg.SendBatchSize = 100
+	cfg.SendBatchMaxSize = 100
+	cfg.Timeout = 3 * time.Second
+	cfg.MaxInFlightBytesMiB = 2
+	creationSet := processortest.NewNopCreateSettings()
+	creationSet.MetricsLevel = configtelemetry.LevelDetailed
+	requestCount := 50
+	// keep spansPerRequest small to ensure multiple contexts end up in the same batch.
+	spansPerRequest := 5
+	exp := tracetest.NewInMemoryExporter()
+	tp := sdktrace.NewTracerProvider(
+		sdktrace.WithBatcher(exp),
+	)
+	otel.SetTracerProvider(tp)
+	tracer := tp.Tracer("otel")
+	bg, rootSp := tracer.Start(context.Background(), "test_start_parent")
+	rootSp.End()
+
+	createSet := exporter.CreateSettings{
+		TelemetrySettings: component.TelemetrySettings{
+			TracerProvider: tp,
+			Logger: zap.NewNop(),
+		},
+	}
+	opt := exporterhelper.WithQueue(exporterhelper.QueueSettings{
+		Enabled: false,
+	})
+	next, err := exporterhelper.NewTracesExporter(bg, createSet, Config{}, func (ctx context.Context, td ptrace.Traces) error {return nil}, opt)
+	require.NoError(t, err)
+
+	bp, err := newBatchTracesProcessor(creationSet, next, cfg, true)
+	require.NoError(t, err)
+	require.NoError(t, bp.Start(context.Background(), componenttest.NewNopHost()))
+
+	callCtxs := []context.Context{
+		bg,
+		client.NewContext(bg, client.Info{
+			Metadata: client.NewMetadata(map[string][]string{
+				"token1": {"single"},
+				"token3": {"n/a"},
+			}),
+		}),
```

Comment:
The map values here are irrelevant, is that right? The use of client.NewContext makes non-same contexts, but these valuesa re otherwise not tested I think. If so, probably an empty map will also work?